core/src/main/java/nl/nn/adapterframework/receivers/PullingListenerContainer.java

```diff
@@ -280,38 +286,65 @@ public void run() {
 								pollToken.release();
 							}
 						}
+
 						try {
-							receiver.processRawMessage(listener, rawMessage, threadContext);
+							if (receiver.getMaxRetries()>=0) {
+								messageId = listener.getIdFromRawMessage(rawMessage, threadContext);
+								String correlationId = (String) threadContext.get(PipeLineSession.technicalCorrelationIdKey);
+								final M rawMessageFinal = rawMessage;
+								final Map<String,Object> threadContextFinal = threadContext;
+								if (receiver.hasProblematicHistory(messageId, false, rawMessage, () -> listener.extractMessage(rawMessageFinal, threadContextFinal), threadContext, correlationId) ) {
+									tooManyRetries = true;
+								}
+							}
+							receiver.processRawMessage(listener, rawMessage, threadContext, true);
+							successfullyProcessed = true;
 							if (txStatus != null) {
 								if (txStatus.isRollbackOnly()) {
+									successfullyProcessed = false;
 									receiver.warn("pipeline processing ended with status RollbackOnly, so rolling back transaction");
 									rollBack(txStatus, rawMessage, "Pipeline processing ended with status RollbackOnly");
 								} else {
 									txManager.commit(txStatus);
 								}
+								txStatus = null;
 							}
 						} catch (Exception e) {
+							receiver.error("caught Exception processing message", e);
 							try {
 								if (txStatus != null && !txStatus.isCompleted()) {
+									successfullyProcessed = false;
 									rollBack(txStatus, rawMessage, "Exception caught ("+e.getClass().getTypeName()+"): "+e.getMessage());
+									txStatus = null;
 								}
 							} catch (Exception e2) {
 								receiver.error("caught Exception rolling back transaction after catching Exception", e2);
-							} finally {
-								if (receiver.isOnErrorContinue()) {
-									receiver.error("caught Exception processing message, will continue processing next message", e);
-								} else {
-									receiver.exceptionThrown("exception occured while processing message", e); //actually use ON_ERROR and don't just stop the receiver
-								}
+							}
```

Comment:
deze finally zorgt er voor dat ongeacht of de rollback goed/fout gaat hij de melding geeft. is het handig dit weg te halen?